test/libyul/YulOptimizerTest.h

```diff
@@ -73,6 +73,7 @@ class YulOptimizerTest: public solidity::frontend::test::EVMVersionRestrictedTes
 	std::string m_expectation;
 
 	Dialect const* m_dialect = nullptr;
+	std::string m_step;
```

Comment:
What is the difference between `m_step` and `m_optimizerStep`?

Reply:
`m_optimizerStep` and `m_step` seem to be essentially the same thing.
The first one comes from the file path, the second one comes from a setting.

**Note**: it might be the only situation when the test behavior is controlled not just by the file content, but by the file path too.

We may want to get rid of the `m_step` setting, and instead write the step to the expectations.
So we would still be able to see the optimization step in the file itself.

Reply:
Please either remove one of them or at least add a comment that explains this.